spyder/api/plugins/new_api.py

```diff
@@ -781,302 +1211,444 @@ def get_command_line_options(self):
+    List of file extensions which the plugin can open.
+
+    If the user opens a file with one of these extensions, then the file
+    will open in this plugin using its :meth:`open_file` method.
+
+    Examples
+    --------
+
+    For example, in the Spyder-Notebook plugin to open Jupyter notebooks:
+
+    .. code-block:: python
+
+        FILE_EXTENSIONS = [".ipynb"]
+    """
+
+    CAN_HANDLE_EDIT_ACTIONS: bool = False
+    """
+    Whether the plugin can handle editing actions.
+
+    If set to ``True``, then the :meth:`undo`, :meth:`redo`, :meth:`cut`,
+    :meth:`copy`, :meth:`paste` and :meth:`select_all` methods will be called
+    to handle the corresponding actions.
+
+    If set to ``False``, the default, the corresponding actions fall back
+    to calling these methods on the :guilabel:`Editor`.
+
+    Individual actions can be disabled with the
+    :meth:`spyder.plugins.application.plugin.enable_edit_action` method
+    in the :guilabel:`Application` plugin.
+    """
+
     CAN_HANDLE_SEARCH_ACTIONS = False
+    """
+    Whether the plugin can handle search actions.
+
+    If set to ``True``, then the :meth:`find`, :meth:`find_next`,
+    :meth:`find_previous` and :meth:`replace` methods will be called
+    to handle the corresponding actions.
+
+    If set to ``False``, the default, the corresponding actions fall back
+    to calling these methods on the :guilabel:`Editor`.
+
+    Individual actions can be disabled with the
+    :meth:`spyder.plugins.application.plugin.enable_search_action` method
+    in the :guilabel:`Application` plugin.
+    """
 
     # ---- API: Available signals
     # -------------------------------------------------------------------------
     sig_focus_changed = Signal()
     """
-    This signal is emitted to inform the focus of this plugin has changed.
+    Report the focus state of this plugin has changed.
     """
 
     sig_toggle_view_changed = Signal(bool)
     """
-    This action is emitted to inform the visibility of a dockable plugin
-    has changed.
+    Report that visibility of a dockable plugin has changed.
 
     This is triggered by checking/unchecking the entry for a pane in the
-    `Window > Panes` menu.
+    :menuselection:`Window --> Panes` menu.
 
     Parameters
     ----------
     visible: bool
-        New visibility of the dockwidget.
+        Whether the dockwidget has been shown (``True``) or hidden (``False``).
     """
 
     sig_switch_to_plugin_requested = Signal(object, bool)
     """
-    This signal can be emitted to inform the main window that this plugin
-    requested to be displayed.
+    Request the main window show this plugin's widget.
 
-    Notes
-    -----
-    This is automatically connected to main container/widget at plugin's
+    This is automatically connected to main container/widget at plugin
     registration.
+
+    Parameters
+    ----------
+    plugin: SpyderDockablePlugin
+        The plugin object to show.
+    force_focus: bool | None
```

Comment:
```suggestion
    force_focus: bool
```

The signature of the signal doesn't allow for this to be `None`.